### PR TITLE
Add http request example to rp

### PR DESCRIPTION
**Pull request by 1-rafael-1**
After myself and 2 others had the same question in 3 days on matrix: Maybe You want an example?

This is not strictly an embassy thing, but it took me (= new to Rust and fairly new to embedded) quite some time & a lot of kind help on matrix to figure out. So my guess is others might benefit.
(thanks @jamesmunns especially!)

Word of caution: Binary size grew > 10xxkb and flashing that somehow corrupts the device and the code fails. The exact same code copied into my own project works just fine. So I added a secction to `cargo.toml` to reduce binary size:

```toml
[profile.dev]
lto = true
opt-level = "z"
```

That way the code now runs without issue, with less than half the original binary size. 

-> if that is not okay, then the entire thing should not go into the repo without some other solution. As a newbie I would then need help to do it differently.

### CODE REVIEW

**Comment by 1-rafael-1** [comment]
I tested some random other examples and it does not look like I broke stuff.

File: examples/rp/src/bin/wifi_webrequest.rs
Comment (jamesmunns) [inline examples/rp/src/bin/wifi_webrequest.rs:1]: This file has some commented out code, could you add comments why it needs to be there, or remove it?

Comment (1-rafael-1) [reply examples/rp/src/bin/wifi_webrequest.rs:1]: sure, done :-)

**Review by jamesmunns — APPROVED** [review]
Looks good to me, let's see what CI thinks :)

**Pull request merged**